sound/soc/sof/sof-acpi-dev.c

```diff
@@ -133,7 +133,12 @@ static void sof_acpi_probe_complete(struct device *dev)
 	/* allow runtime_pm */
 	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY_MS);
 	pm_runtime_use_autosuspend(dev);
+	pm_runtime_set_active(dev);
 	pm_runtime_enable(dev);
+	pm_runtime_mark_last_busy(dev);
+
+	/* put_autosuspend doesn't seem to work, so use put_noidle */
+	pm_runtime_put_noidle(dev);
```

Comment:
for Soundwire we use a different sequence

		pm_runtime_set_autosuspend_delay(&md->dev, 3000);
		pm_runtime_use_autosuspend(&md->dev);
		pm_runtime_mark_last_busy(&md->dev);

		pm_runtime_set_active(&md->dev);
		pm_runtime_enable(&md->dev);
	
		pm_runtime_idle(&md->dev);

I am not sure what you would do a put_noidle and use a get_noresume, this doesn't look very useful to me to do a put if you haven't done a get first. This is different from PCI where the core does play with refcounts.

Reply:
@plbossart this one is very confusing for me. I must have played with all the options but for pm_runtime_idle(). let me try that

Reply:
@plbossart I fixed this. I problems I was seeing what not because of this after all.